File: Selenium/Python/searchAndFillForm/RemoteWebDriverTest.py
import time

from selenium.webdriver import ActionChains
from selenium.webdriver.remote.command import Command
from selenium.webdriver.remote.mobile import Mobile

from WindTunnelUtils import WindTunnelUtils
from PerfectoLabUtils import PerfectoLabUtils
from selenium import webdriver
from selenium.webdriver.common.by import By

class RemoteWebDriverTest:

    def __init__(self):
        pass

    @staticmethod
    def main():
        print('Run started')
        
		""" Provide over here your cloud information : Host , user and password """
        host        = "My_Host.perfectomobile.com" 
        user        = "My_User" 
        password    = "My_Password" 
        
		#Capabilities for example , Provide your device capabilities here.
        capabilities = {
            'user'              : user,
            'password'          : password,
            'platformName'      : 'Windows',
            'platformVersion'   : '8.1',
            'browserName'       : 'Chrome',
            'browserVersion'    : '49'
        }


        driver = webdriver.Remote("https://" + host + "/nexperience/perfectomobile/wd/hub", capabilities)
        driver.implicitly_wait(15)
        driver.maximize_window()

		""" Test """
        try:
            print('Test started')
			
            #Search in google.
            driver.get('https://google.com')
            driver.find_element(By.NAME, 'q').send_keys('perfecto mobile')
            driver.find_element(By.XPATH, "//*[@id='rso']/div[1]/div[1]/div/h3/a").click() # Clicks the first search result.
            
            #Fill a form in perfecto site.
            driver.find_element(By.XPATH, "//*[text() = 'Start Free']").click()
            driver.find_element(By.ID , "FirstName").send_keys('MyFirstName')
            driver.find_element(By.ID , "LastName").send_keys('MyLastName')
            driver.find_element(By.ID , "Company").send_keys('MyCompany');
            driver.find_element(By.ID , "Mobile_Testing_Role__c").click()
            driver.find_element(By.XPATH , "//*[@value = 'Development']").click()
            driver.find_element(By.ID , "Email").send_keys("MyEmail@Somehost.com")
            driver.find_element(By.ID , "Phone").send_keys("123456789")
            driver.find_element(By.ID , "Country").click()
            driver.find_element(By.XPATH , "//*[@value = 'Israel']").click()

        except Exception as e:
            print(e)
            
        finally:
            try:
                # disconnect from the Remote server
                if driver is not None:
                    # Retrieve the URL of the Wind Tunnel Report, can be saved to your execution summary and used to download the report at a later point
                    report_url = driver.capabilities[WindTunnelUtils.SINGLE_TEST_REPORT_URL_CAPABILITY]

                    driver.close()

                    # In case you want to download the report or the report attachments, do it here.
                    # PerfectoLabUtils.download_report(driver, 'pdf', '/test/report')
                    # PerfectoLabUtils.download_attachment(driver, 'video', '/test/video', 'flv')
                    # PerfectoLabUtils.download_attachment(driver, 'image', '/test/image', 'jpg')

            except Exception as e:
                print(e)

            driver.quit()

        print('Run ended')

    @staticmethod
    def switch_to_context(driver, context):
        driver.execute(Command.SWITCH_TO_CONTEXT, {"name": context})

    @staticmethod
    def get_current_context_handle(driver):
        mobile = Mobile(driver)
        return mobile.context

    @staticmethod
    def get_context_handles(driver):
        mobile = Mobile(driver)
        return mobile.contexts

if __name__ == "__main__":
    RemoteWebDriverTest().main()

File: Selenium/C#/searchAndFillForm/RemoteWebDriverTest.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Diagnostics;
using System.Threading;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Remote;

namespace GoogleSearchAndFillForm
{

    [TestClass]
    public class RemoteWebDriverTest
    {
        private RemoteWebDriverExtended driver;

        [TestInitialize]
        public void PerfectoOpenConnection()
        {
        	
            var browserName = "mobileOS";
	    //TODO: Provide you cloud host, user and password.
            var host = "My_Host.perfectomobile.com";
	    var user = "My_User";
	    var pass = "My_Password";

            DesiredCapabilities capabilities = new DesiredCapabilities(browserName, string.Empty, new Platform(PlatformType.Any));
            capabilities.SetCapability("user", user);
            capabilities.SetCapability("password", pass);

            //TODO: Provide your device capabilities.
			//Capabilities for example: 
            capabilities.SetCapability("platformName", "Windows");
            capabilities.SetCapability("platformVersion", "7");
            capabilities.SetCapability("browserName", "Chrome");
            capabilities.SetCapability("browserVersion", "49");

            capabilities.SetPerfectoLabExecutionId(host);

            var url = new Uri(string.Format("http://{0}/nexperience/perfectomobile/wd/hub", host));
            driver = new RemoteWebDriverExtended(new HttpAuthenticatedCommandExecutor(url), capabilities);

            driver.Manage().Timeouts().ImplicitlyWait(TimeSpan.FromSeconds(15));
            driver.Manage().Window.Maximize();
        }

        /**
        * scenario:
        * 1. Open https://google.com .
        * 2. Type perfectomobile and click search.
        * 3. Press on the first search result.
        * 4. Click on START FREE btn on perfecto site.
        * 5. Fill the form and finish.
        **/
        [TestMethod]
        public void WebDriverTestMethod()
        {
            try
            {
                //Search in google.
                driver.Navigate().GoToUrl("https://google.com");
                driver.FindElementByName("q").SendKeys("perfecto mobile"); //Insert perfecto mobile.
                driver.FindElementByXPath("//*[@id='rso']/div[1]/div[1]/div/h3/a").Click();// Click the first search result.

                //Fills a form on perfecto website.
                driver.FindElementByXPath("//*[text() = 'Start Free']").Click();
                driver.FindElementById("FirstName").SendKeys("MyFirstName");
                driver.FindElementById("LastName").SendKeys("MyLaseName");
                driver.FindElementById("Company").SendKeys("MyCompany");
                driver.FindElementById("Mobile_Testing_Role__c").Click();
                driver.FindElementByXPath("//*[@value = 'Development']").Click();
                driver.FindElementById("Email").SendKeys("MyEmail@Somehost.com");
                driver.FindElementById("Phone").SendKeys("123456789");
                driver.FindElementById("Country").Click();
                driver.FindElementByXPath("//*[@value = 'Israel']").Click();
            }
            catch (Exception ex)
            {
                Trace.WriteLine(string.Format("Error getting test logs: {0}", ex.Message));
            }
        }

        [TestCleanup]
        public void PerfectoCloseConnection()
        {
            // Retrieve the URL of the Single Test Report, can be saved to your execution summary and used to download the report at a later point
            string reportUrl = (string)(driver.Capabilities.GetCapability(WindTunnelUtils.SINGLE_TEST_REPORT_URL_CAPABILITY));

            driver.Close();

            // In case you want to download the report or the report attachments, do it here.
            //try
            //{
            //    driver.DownloadReport(DownloadReportTypes.pdf, "C:\\test\\report");
            //    driver.DownloadAttachment(DownloadAttachmentTypes.video, "C:\\test\\report\\video", "flv");
            //    driver.DownloadAttachment(DownloadAttachmentTypes.image, "C:\\test\\report\\images", "jpg");
            //}
            //catch (Exception ex)
            //{
            //    Trace.WriteLine(string.Format("Error getting test logs: {0}", ex.Message));
            //}

            driver.Quit();
        }


    }
}
